sdk/python/lib/pulumi/runtime/mocks.py

```diff
@@ -105,13 +105,17 @@ def ReadResource(self, request):
         return resource_pb2.ReadResourceResponse(urn=urn, properties=props_proto)
 
     def RegisterResource(self, request):
+        urn = self.make_urn(request.parent, request.type, request.name)
+
+        if request.type == "pulumi:pulumi:Stack":
+            return resource_pb2.RegisterResourceResponse(urn=urn)
```

Comment:
This is consistent with what we do for Go:

https://github.com/pulumi/pulumi/blob/7b17463031d3e4a7d31764c8d5811985855820c1/sdk/go/pulumi/mocks.go#L108-L112

Interestingly, C# does not have a check like this. Currently, `IMocks.NewResourceAsync` is called for `"pulumi:pulumi:Stack"`. I think this is a bug that we should fix to be consistent with the other languages. If you agree, I'll open an issue.

Reply:
Yes, I think you are right

Reply:
https://github.com/pulumi/pulumi/issues/4524